### PR TITLE
add batch size changes

### PR DESCRIPTION
**Pull request by ankitchiplunkar**

### CODE REVIEW

File: contracts/VolOracle.sol [hunk 1]
```diff
@@ -129,9 +141,9 @@ contract VolOracle {
                 tickSquareDelta + prevObservation.tickSquareCumulative
             );
         }
-
+        (uint32 lastBlockTimestamp, , , ) = uniPool.observations(endIndex % poolCardinality);
         volOracleState.observationIndex = volObservationIndex;
         volOracleState.lastObservationIndex = endIndex % poolCardinality;
-        volOracleState.lastBlockTimestamp = block.timestamp;
+        volOracleState.lastBlockTimestamp = uint256(lastBlockTimestamp);
```
Comment (PangZhi) [inline contracts/VolOracle.sol:148]: we also change the lastBlockTimestamp value when we initialize the pool. To keep it consistent, we'd better update there as well

Comment (PangZhi) [reply contracts/VolOracle.sol:148]: Let's update the comment of this field in the struct as it has new meaning now

**Comment by PangZhi** [comment]
maybe we should also change the fillInobservations function to return a boolean value to indicate whether it has next batch, just like how pagination works. Otherwise users may not know whether they need to continue to call fillInobservations.

**Comment by ankitchiplunkar** [comment]
converted it into a notion card: https://www.notion.so/a04752eb587047c685651f9e1d8209d1?v=8079521e5b6d493dae47609c13bad950&p=ee7c8d5885224d109bd4db2a9e9fbf1e&pm=s

**Review by PangZhi — APPROVED** [review]

**Review by PangZhi — APPROVED** [review]

**Pull request merged**